modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/IgniteLimit.java

```diff
@@ -163,7 +163,7 @@ else if (!requiredCollation.satisfies(relCollation))
     /**
      * @return Integer value of the literal expression.
      */
-    private double doubleFromRex(RexNode n, double def) {
+    static double doubleFromRex(RexNode n, double def) {
```

Comment:
Let's move it to `RexUtils` class. Also, javadoc and perhaps implementation should be fixed (I don't understand why we convert it to Integer.class first).

Reply:
Fixed